documentation/API-design-guidelines.md

```diff
@@ -793,9 +793,9 @@ The Following table compiles the guidelines to be adopted:
 |     1      | None of the provided device identifiers is supported by the implementation |       422        | UNSUPPORTED_DEVICE_IDENTIFIERS | phoneNumber is required.                                 |
 |     2      | Some identifier cannot be matched to a device                              |       404        |        DEVICE_NOT_FOUND        | Device identifier not found.                             |  
 |     3      | Device identifiers mismatch                                                |       422        |  DEVICE_IDENTIFIERS_MISMATCH   | Provided device identifiers are not consistent.          |
-|     4      | Invalid access token context                                               |       403        |     INVALID_TOKEN_CONTEXT      | Device identifiers are not consistent with access token. |
+|     4      | An explicit identifier is provided when a device or phone number has already been identified from the access token |       422        | UNNECESSARY_IDENTIFIER  | The device is already identified by the access token. |
```

Comment:
Do we want to throw an error even if the device identifier matches? I agree that the consumer did not need to send it, but given that it is a match,  I wouldn't throw an error for this and make it send the request again.

Reply:
Yes, this is necessary. See Issue [#259](https://github.com/camaraproject/Commonalities/issues/259), and the summary in the PR description above.

To only throw an error on a mismatch would be enabling "number verification" functionality for all APIs that allow authorisation code flow, irrespective of User consent. For CIBA, it would allow confirmation as to whether an IP address or operator token matched a particular MSISDN.

The alternative was to always ignore the explicit identifier if a 3-legged token was used (irrespective of whether the User matched or not). The problem with that solution is that the API consumer might misunderstand which User is the subject of the API if they were not aware they were using a 3-legged token which already had an associated User.